iceoryx_utils/test/moduletests/test_cxx_newtype.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_utils/cxx/newtype.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox;
+using namespace iox::cxx;
+
+TEST(NewType, ComparableDoesCompile)
+{
+    cxx::NewType<int, newtype::ConstructByValueCopy, newtype::Comparable> a(123), b(456);
```

Comment:
Could you maybe also create a struct here? Just to not have examples where it is not used as intended.

Reply:
done